biz.aQute.bndlib/src/aQute/bnd/plugin/jpms/JPMSModuleInfoPlugin.java

```diff
@@ -185,25 +192,25 @@ public void verify(final Analyzer analyzer) throws Exception {
 			}
 			MapStream.of(jar.getDirectories())
 				.filter((path, resources) -> (resources != null) && !resources.isEmpty() && !path.isEmpty())
-				.keys()
-				.map(analyzer::getPackageRef)
-				.filter(PackageRef::isValidPackageName)
-				.forEachOrdered(packageRef -> index.put(packageRef, new Attrs(attrs)));
+				.map((k, v) -> MapStream.entry(analyzer.getPackageRef(k), imports.get(analyzer.getPackageRef(k))))
+				.filterKey(PackageRef::isValidPackageName)
+				.forEach((packageRef, packageAttrs) -> index.put(packageRef, new Attrs(packageAttrs, attrs)));
 		}
 
-		ModuleInfoBuilder builder = nameAccessAndVersion(moduleInstructions, requireCapabilities, analyzer);
+		Set<PackageRef> bcpEntries = MapStream.of(analyzer.getBundleClassPathTypes())
+			.mapKey(TypeRef::getPackageRef)
+			.keys()
+			.collect(toSet());
```

Comment:
```suggestion
		Set<PackageRef> bcpEntries = analyzer.getBundleClassPathTypes()
			.keySet()
			.map(TypeRef::getPackageRef)			  
			.collect(toSet());
```

You only need the keys, so you can avoid MapStream here.

Reply:
fixed, please see changes.